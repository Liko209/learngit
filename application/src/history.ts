import _ from 'lodash';
import {
  LocationDescriptorObject,
  Path,
  LocationState,
  createBrowserHistory,
  createMemoryHistory,
} from 'history';

const isTestENV = process && process.env.NODE_ENV === 'test';
const history = isTestENV ? createMemoryHistory() : createBrowserHistory();
const historyPush = history.push;
const historyReplace = history.replace;

const isSameLocation = (
  options: LocationDescriptorObject | Path,
  newState?: LocationState,
) => {
  const { pathname, search, hash, state } = history.location;
  if (
    typeof options === 'string' &&
    (`${pathname}${search}${hash}` !== options || !_.isEqual(newState, state))
  ) {
    return false;
  }
  if (
    typeof options === 'object' &&
    (pathname !== options.pathname ||
      search !== options.search ||
      hash !== options.hash ||
      !_.isEqual(state, options.state))
  ) {
    return false;
  }
  return true;
};

const pushOrReplace = (action: Function) => {
  return (options: LocationDescriptorObject | Path, state?: LocationState) => {
    if (!isSameLocation(options, state)) {
      action(options, state);
    }
  };
};

history.push = pushOrReplace(historyPush);
history.replace = pushOrReplace(historyReplace);

export { isSameLocation, pushOrReplace };

export default history;
