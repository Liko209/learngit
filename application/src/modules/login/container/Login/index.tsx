/*
 * @Author: Devin Lin (devin.lin@ringcentral.com)
 * @Date: 2018-08-03 13:59:53
 * Copyright Â© RingCentral. All rights reserved.
 */

import React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { AccountService } from 'sdk/module/account';
import { ServiceLoader, ServiceConfig } from 'sdk/module/serviceLoader';

interface IProps extends RouteComponentProps<any> {}

class Login extends React.Component<IProps> {
  constructor(props: IProps) {
    super(props);
    this.onClick = this.onClick.bind(this);
    this._checkIfLogin();
  }

  private _checkIfLogin() {
    const accountService = ServiceLoader.getInstance<AccountService>(
      ServiceConfig.ACCOUNT_SERVICE,
    );
    if (accountService.isLoggedIn()) {
      const { history } = this.props;
      history.replace('/messages');
    }
  }
  onClick() {
    const { location, history } = this.props;
    localStorage.setItem('login', 'true'); // todo
    history.replace((location.state && location.state.from) || '/');
  }

  render() {
    return (
      <div>
        login page
        <button onClick={this.onClick} type='button'>
          Login
        </button>
      </div>
    );
  }
}

export default Login;
