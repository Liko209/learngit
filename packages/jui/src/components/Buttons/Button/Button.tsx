/*
 * @Author: Chris Zhan (chris.zhan@ringcentral.com)
 * @Date: 2018-08-22 15:24:02
 * Copyright Â© RingCentral. All rights reserved.
 */
import * as React from 'react';
import MuiButton, {
  ButtonProps as MuiButtonProps,
} from '@material-ui/core/Button';
import { RuiCircularProgress } from 'rcui/components/Progress';
import { Palette, Theme } from '../../../foundation/theme/theme';
import styled from '../../../foundation/styled-components';
import {
  typography,
  spacing,
  palette,
  width,
  rippleEnter,
  height,
  grey,
  primary,
} from '../../../foundation/utils/styles';
import { Omit } from '../../../foundation/utils/typeHelper';

type Variant = 'text' | 'contained' | 'outlined';

type JuiButtonColor = 'primary' | 'secondary' | 'negative';

type JuiButtonProps = Omit<MuiButtonProps, 'innerRef' | 'variant' | 'color'> & {
  size?: 'small' | 'large';
  variant?: Variant;
  disabled?: boolean;
  color?: JuiButtonColor;
  loading?: boolean;
  component?: React.ElementType;
};

const ColorMap: {
  [x: string]: [keyof Palette, string];
} = {
  primary: ['primary', 'main'],
  secondary: ['secondary', 'main'],
  negative: ['semantic', 'negative'],
};

const touchRippleClasses = {
  rippleVisible: 'rippleVisible',
};

const WrappedMuiButton = (props: JuiButtonProps) => {
  const { variant, color, children, loading, ...restProps } = props;
  return (
    <MuiButton
      classes={{
        disabled: 'disabled',
        contained: 'containedButtonStyle',
        text: 'textButtonStyle',
        outlined: 'outlineButtonStyle',
      }}
      TouchRippleProps={{ classes: touchRippleClasses }}
      variant={variant}
      {...restProps}
    >
      {loading ? <RuiCircularProgress size={16} color="inherit" /> : children}
    </MuiButton>
  );
};

const ButtonColor = ({
  theme,
  color = 'primary',
}: {
  theme: Theme;
  color: JuiButtonColor;
}) =>
  theme.palette.getContrastText(
    palette(ColorMap[color][0], ColorMap[color][1])({ theme }),
  );
const StyledButton = styled<JuiButtonProps>(WrappedMuiButton)`
  && {
    min-width: ${({ theme }) => width(26)({ theme })};
    padding: ${spacing(2.5, 4)};
    display: flex;
    text-transform: none;
    ${typography('button')};
    color: ${palette('primary', 'main')};
    text-align: center;
    box-shadow: unset;
    &.containedButtonStyle {
      color: ${ButtonColor};
      background-color: ${({ color = 'primary' }) =>
        palette(ColorMap[color][0], ColorMap[color][1])};
      &:hover {
        opacity: ${({ theme }) => 1 - theme.palette.action.hoverOpacity};
      }
      &.disabled {
        background-color: ${({ theme, loading }) =>
          loading ? '' : palette('accent', 'ash')({ theme })};
        color: ${ButtonColor};
      }
    }

    &.textButtonStyle {
      &.disabled {
        color: ${({ theme, loading }) =>
          loading ? '' : palette('accent', 'ash')({ theme })};
      }
      &:hover {
        background-color: ${palette('primary', 'main', 1)};
      }
    }

    &.outlineButtonStyle {
      border-color: ${palette('primary', 'main')};
    }

    &.roundButtonStyle {
      height: ${({ theme }) => height(7)({ theme })};
      border-radius: ${({ theme }) => spacing(7)({ theme })};
      padding: ${({ theme }) => spacing(0, 4)({ theme })};
      background-color: ${({ theme }) =>
        theme.palette.getContrastText(primary('700')({ theme }))};
      color: ${primary('700')};
      ${typography('caption1')};
      min-height: unset;
      width: inherit;
      &:hover {
        background-color: ${grey('50')};
        &:active {
          background-color: ${grey('100')};
        }
      }

      .rippleVisible {
        opacity: ${({ theme }) => theme.palette.action.hoverOpacity * 2};
        transform: scale(1);
        animation-name: ${({ theme }) => rippleEnter(theme)};
      }
    }
  }
`;

const JuiButtonComponent: React.StatelessComponent<JuiButtonProps> = (
  props: JuiButtonProps,
) => <StyledButton {...props} />;

JuiButtonComponent.defaultProps = {
  size: 'large',
  color: 'primary',
  variant: 'contained',
  disabled: false,
};

const JuiButton = styled(React.memo(JuiButtonComponent))``;

export { JuiButton, JuiButtonProps, JuiButtonColor, ColorMap };
