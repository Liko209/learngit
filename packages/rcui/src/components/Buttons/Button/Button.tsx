import * as React from 'react';
import MuiButton, {
  ButtonProps as MuiButtonProps,
} from '@material-ui/core/Button';
import { fade } from '@material-ui/core/styles/colorManipulator';
import styled from '../../../foundation/styled-components';
import { Omit } from '../../../foundation/utils/typeHelper';
import {
  spacing,
  palette,
  typography,
  shadows,
} from '../../../foundation/shared/theme';
import dataColors from '../../../foundation/shared/colors.json';
import { RuiCircularProgress } from '../../Progress';
import { rippleEnter } from '../../../foundation/shared/styles';

type RuiButtonVariant = MuiButtonProps['variant'] | 'round';

type RuiButtonColor = MuiButtonProps['color'] | 'error';

type RuiButtonSize = 'medium' | 'large' | 'small';

type RuiButtonProps = {
  size?: RuiButtonSize;
  variant?: RuiButtonVariant;
  disabled?: boolean;
  color?: RuiButtonColor;
  loading?: boolean;
} & Omit<MuiButtonProps, 'innerRef' | 'color' | 'variant'>;

const ColorMap = {
  primary: palette('primary', 'main'),
  secondary: palette('secondary', 'main'),
  error: palette('error', 'main'),
};

const touchRippleClasses = {
  rippleVisible: 'rippleVisible',
};

const sizes = {
  large: '36px',
  medium: '32px',
};

const minWith = {
  large: '64px',
  medium: '40px',
};

const borderColors = {
  primary: palette('primary', 'main'),
  error: palette('error', 'main'),
  secondary: palette('secondary', 'main'),
};

const roundButtonPadding = {
  large: '17px',
  medium: '13px',
};

const RuiButton = (props: RuiButtonProps) => {
  const {
    variant,
    color,
    children,
    loading,
    fullWidth,
    size,
    ...restProps
  } = props;
  const _color = color;
  let _size = size;
  if (fullWidth) {
    _size = 'medium';
  }
  let _variant = variant;
  const classes = {
    disabled: 'disabled',
    contained: 'containedButtonStyle',
    text: 'textButtonStyle',
    outlined: 'outlineButtonStyle',
    root: 'root',
    label: 'label',
    fab: 'roundButtonStyle',
  };
  if (variant === 'round') {
    _variant = 'fab';
    restProps.disableRipple = true;
    restProps.disabled = false;
  }
  const loadingCom = loading ? (
    <RuiCircularProgress size={20} />
  ) : (
      children
    );
  return (
    <StyledButton
      classes={classes}
      TouchRippleProps={{ classes: touchRippleClasses }}
      variant={_variant}
      color={_color}
      size={_size}
      fullWidth={fullWidth}
      {...restProps}
    >
      {loadingCom}
    </StyledButton>
  );
};

const StyledButton = styled(MuiButton) <RuiButtonProps>`
  && {
    padding-right: ${spacing('s')};
    padding-left: ${spacing('s')};
    ${typography('button')};
    color: ${palette('primary', 'main')};
    height: ${({ size = 'large' }) => sizes[size]};
    & > span[class*="label"] {
      min-width: ${({ size = 'large' }) => minWith[size]};
    }
    &.root {
      text-transform: none;
      padding-top: 0;
      padding-bottom: 0;
    }
    &.containedButtonStyle {
      color: white;
      background-color: ${({ color = 'primary' }) => ColorMap[color]};
      box-shadow: unset;
      &:active {
      }
      &:hover {
        opacity: ${({ theme }) => 1 - theme.palette.action.hoverOpacity};
      }
      &:disabled {
        color: white;
        box-shadow: none;
        background-color: ${palette('action', 'disabledBackground')};
      }
    }
    &.textButtonStyle {
      color: ${({ color = 'primary' }) => ColorMap[color]};
      &.disabled {
        color: ${dataColors['ash']};
      }
      &:hover {
        background-color: ${({ theme, color }) =>
    fade(
      theme.palette[`${color}`].main,
      theme.palette.action.hoverOpacity,
    )};
      }
    }
    &.outlineButtonStyle {
      border-color: ${({ color = 'primary' }) => borderColors[color]};
      background-color: ${palette('common', 'white')};
      color: ${({ color = 'primary' }) => ColorMap[color]};
      &.disabled {
        color: ${palette('action', 'disabled')};
        background-color: ${palette('common', 'white')};
        border-color: ${palette('action', 'disabledBackground')};
      }
      &:hover {
        background-color: ${({ theme, color }) =>
    fade(
      theme.palette[`${color}`].main,
      theme.palette.action.hoverOpacity,
    )};
      }
    }
    &.roundButtonStyle {
      height: ${({ size = 'large' }) => sizes[size]};
      border-radius: 28px;
      padding: 0 ${({ size = 'large' }) => roundButtonPadding[size]};
      background-color: ${palette('common', 'white')};
      color: ${({ theme }) => theme.palette.primary.main};
      ${typography('caption1')};
      min-height: unset;
      width: inherit;
      box-shadow: ${shadows(3)};
      &:hover {
        background-color: ${({ theme }) => theme.palette.grey[50]};
        &:active {
          background-color: ${({ theme }) => theme.palette.grey[100]};
        }
      }
      .rippleVisible {
        opacity: ${({ theme }) => theme.palette.action.hoverOpacity * 2};
        transform: scale(1);
        animation-name: ${({ theme }) => rippleEnter(theme)};
      }
    }
  }
` as React.ComponentType<RuiButtonProps>;

RuiButton.defaultProps = {
  size: 'large',
  color: 'primary',
  variant: 'contained',
  disabled: false,
};
export { RuiButton, RuiButtonProps, RuiButtonColor };
