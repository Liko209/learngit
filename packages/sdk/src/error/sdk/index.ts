export * from './types';
export * from './JSdkError';
