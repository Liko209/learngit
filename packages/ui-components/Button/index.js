var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
import React from 'react';
import styled, { withTheme } from 'styled-components';
import MuiButton from '@material-ui/core/Button';
export var CustomButton = function (props) {
    return React.createElement(MuiButton, __assign({}, props));
};
export var Button = styled(withTheme(CustomButton)).attrs({})(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  min-width: 104px !important;\n  min-height: ", " !important;\n  height: ", " !important;\n  padding: ", " !important;\n  font-size: 14px !important;\n  line-height: 16px !important;\n  border-radius: 2px !important;\n"], ["\n  min-width: 104px !important;\n  min-height: ",
    " !important;\n  height: ",
    " !important;\n  padding: ", " !important;\n  font-size: 14px !important;\n  line-height: 16px !important;\n  border-radius: 2px !important;\n"])), function (props) {
    return props.size && props.size === 'small' ? '28px' : '40px';
}, function (props) {
    return props.size && props.size === 'small' ? '28px' : '40px';
}, function (props) { return (props.href ? '0' : '0 20px'); });
export default Button;
var templateObject_1;
//# sourceMappingURL=index.js.map