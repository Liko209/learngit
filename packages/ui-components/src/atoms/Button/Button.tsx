/*
 * @Author: Chris Zhan (chris.zhan@ringcentral.com)
 * @Date: 2018-08-22 15:24:02
 * Copyright Â© RingCentral. All rights reserved.
 */
import * as React from 'react';
import tinycolor from 'tinycolor2';
import styled, { keyframes, IDependencies } from '../../styled-components';
import MuiButton, { ButtonProps as MuiButtonProps } from '@material-ui/core/Button';
import { typography, spacing, palette, width } from '../../utils/styles';
import { Theme } from '../../theme';

type JuiButtonProps = MuiButtonProps & {
  size?: 'small' | 'large';
  variant?: 'text' | 'contained';
  disabled?: boolean;
  color?: 'primary' | 'secondary';
};

const rippleEnter = (theme: Theme) => keyframes`
  from {
    transform: scale(0);
    opacity: 0.1;
  }
  to {
    transform: scale(1);
    opacity: ${theme.palette.action.hoverOpacity * 2};
  }
`;

const touchRippleClasses = {
  rippleVisible: 'rippleVisible',
};
const WrappedMuiButton = (props: JuiButtonProps) => (
  <MuiButton
    classes={{ disabled: 'disabled', contained: 'containedButtonStyle' }}
    TouchRippleProps={{ classes: touchRippleClasses }}
    {...props}
  />
);
const StyledButton = styled<JuiButtonProps>(WrappedMuiButton)`
  && {
    min-width: ${({ theme }) => width(26)({ theme })};
    padding-left: ${spacing(4)};
    padding-right: ${spacing(4)};
    ${typography('button')}

    &.containedButtonStyle {
      box-shadow: unset;
      color: white;
      &:hover {
        background-color: ${({ theme, color = 'primary' }) => tinycolor(palette(color, 'main')({ theme })).setAlpha(1 - theme.palette.action.hoverOpacity).toRgbString()};
      }
      &:active {
        box-shadow: unset;
      }
    }

    &.textButtonStyle {
      &.disabled {
        color: ${palette('accent', 'ash')};
      }
    }


    .rippleVisible {
      opacity: ${({ theme }) => theme.palette.action.hoverOpacity * 2};
      transform: scale(1);
      animation-name: ${({ theme }) => rippleEnter(theme)};
    }
  }
`;

type IButton = React.StatelessComponent<JuiButtonProps> & IDependencies;
const JuiButton: IButton = ({ innerRef, ...rest }: JuiButtonProps) => (<StyledButton {...rest} />);

JuiButton.defaultProps = {
  size: 'large',
  color: 'primary',
  variant: 'contained',
  disabled: false,
};

JuiButton.dependencies = [MuiButton];

export { JuiButton, JuiButtonProps };
export default JuiButton;
