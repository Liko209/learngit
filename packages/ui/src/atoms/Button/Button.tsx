/*
 * @Author: Chris Zhan (chris.zhan@ringcentral.com)
 * @Date: 2018-08-22 15:24:02
 * Copyright Â© RingCentral. All rights reserved.
 */
import * as React from 'react';
import styled from '../../styled-components';
import MuiButton, { ButtonProps } from '@material-ui/core/Button';

const StyledButton = styled<ButtonProps>(MuiButton)`
  && {
    min-width: ${({ theme }) => theme.spacing.unit * 26}px;
  }
`;

type IButton = { dependencies?: React.ComponentType[] } & React.SFC<ButtonProps>;

const Button: IButton = ({ innerRef, ...rest }) => {
  return <StyledButton {...rest} />;
};

Button.defaultProps = {
  size: 'large',
};

Button.dependencies = [MuiButton];

export { Button, ButtonProps };
export default Button;
