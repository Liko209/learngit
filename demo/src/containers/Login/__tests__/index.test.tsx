import React from 'react';
import { shallow } from 'enzyme';
import * as H from 'history';
import { service } from 'sdk';

import Login from '../index';
import Button from '../Button';

const { AuthService } = service;

const history = H.createBrowserHistory();
const location = H.createLocation('/');
const match = {
  params: 1,
  isExact: true,
  path: 'login',
  url: 'login',
};
const staticContext: any;
let wrapper: any;

describe('<Login />', () => {
  beforeAll(() => {
    wrapper = shallow(
      <Login
        history={history}
        location={location}
        match={match}
        staticContext={staticContext}
      />,
    );
  });
  describe('render()', () => {
    it('<Login /> should be rendering', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('handleChange()', () => {
    it('username input value should be Devin', () => {
      wrapper
        .find('[name="username"]')
        .simulate('change', { target: { name: 'username', value: 'Devin' } });
      wrapper
        .find('[name="password"]')
        .simulate('change', { target: { name: 'password', value: 'abc123' } });
      expect(wrapper.state('username')).toBe('Devin');
      expect(wrapper.find('[name="username"]').props().value).toBe('Devin');
    });
  });

  describe('handleClick()', () => {
    it('AuthService login() should be called with specific params', () => {
      const authService = {
        login: jest.fn(),
      };
      wrapper
        .find('[name="username"]')
        .simulate('change', { target: { name: 'username', value: 'Devin' } });
      wrapper
        .find('[name="password"]')
        .simulate('change', { target: { name: 'password', value: 'abc123' } });
      AuthService.getInstance = jest.fn().mockReturnValue(authService);
      authService.login = jest.fn().mockResolvedValue(null);
      wrapper.find(Button).simulate('click', { preventDefault: jest.fn });
      expect(authService.login).toHaveBeenCalledWith({
        extension: '',
        password: 'abc123',
        username: 'Devin',
      });
    });
  });
});
