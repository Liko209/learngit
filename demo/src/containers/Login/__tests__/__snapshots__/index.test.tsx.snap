// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> render() <Login /> should be rendering 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    location={
      Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      }
    }
    match={
      Object {
        "isExact": true,
        "params": 1,
        "path": "login",
        "url": "login",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <styled.form>
          <styled.h1>
            <span>
              Sign In
            </span>
          </styled.h1>
          <styled.input
            name="username"
            onChange={[Function]}
            placeholder="Phone number"
            value=""
          />
          <styled.input
            name="extension"
            onChange={[Function]}
            placeholder="Extension"
            value=""
          />
          <styled.input
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
          <styled.button
            disabled={false}
            onClick={[Function]}
          >
            Login
          </styled.button>
          <styled.div>
            <ul />
          </styled.div>
        </styled.form>,
        <Download />,
        <LoginVersionStatus />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.h1>
              <span>
                Sign In
              </span>
            </styled.h1>,
            <styled.input
              name="username"
              onChange={[Function]}
              placeholder="Phone number"
              value=""
            />,
            <styled.input
              name="extension"
              onChange={[Function]}
              placeholder="Extension"
              value=""
            />,
            <styled.input
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />,
            <styled.button
              disabled={false}
              onClick={[Function]}
            >
              Login
            </styled.button>,
            <styled.div>
              <ul />
            </styled.div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <span>
                Sign In
              </span>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign In",
              },
              "ref": null,
              "rendered": "Sign In",
              "type": "span",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "username",
              "onChange": [Function],
              "placeholder": "Phone number",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "extension",
              "onChange": [Function],
              "placeholder": "Extension",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Login",
              "disabled": false,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Login",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <ul />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
              },
              "ref": null,
              "rendered": Array [],
              "type": "ul",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <styled.form>
            <styled.h1>
              <span>
                Sign In
              </span>
            </styled.h1>
            <styled.input
              name="username"
              onChange={[Function]}
              placeholder="Phone number"
              value=""
            />
            <styled.input
              name="extension"
              onChange={[Function]}
              placeholder="Extension"
              value=""
            />
            <styled.input
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
            <styled.button
              disabled={false}
              onClick={[Function]}
            >
              Login
            </styled.button>
            <styled.div>
              <ul />
            </styled.div>
          </styled.form>,
          <Download />,
          <LoginVersionStatus />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.h1>
                <span>
                  Sign In
                </span>
              </styled.h1>,
              <styled.input
                name="username"
                onChange={[Function]}
                placeholder="Phone number"
                value=""
              />,
              <styled.input
                name="extension"
                onChange={[Function]}
                placeholder="Extension"
                value=""
              />,
              <styled.input
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />,
              <styled.button
                disabled={false}
                onClick={[Function]}
              >
                Login
              </styled.button>,
              <styled.div>
                <ul />
              </styled.div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <span>
                  Sign In
                </span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign In",
                },
                "ref": null,
                "rendered": "Sign In",
                "type": "span",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "username",
                "onChange": [Function],
                "placeholder": "Phone number",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "extension",
                "onChange": [Function],
                "placeholder": "Extension",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Login",
                "disabled": false,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Login",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <ul />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [],
                },
                "ref": null,
                "rendered": Array [],
                "type": "ul",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
